src/content/en/2021/caching.md

```diff
@@ -17,4 +19,335 @@ featured_stat_3: TODO
+
+Fun fact the the largest max-age value is 1625540701883604800000 seconds or 51545557517871.8 years (51 trillion).
+
+{{ figure_markup(
+  caption="TODO",
+  content="51 trillion years",
+  classes="big-number",
+  sheets_gid="529870849",
+  sql_file="todo.sql"
+) }}
+
+Talk about `immutable` directive, 12% of desktop requests containing Cache-Control: immutable are on the "static.parastorage.com" host.
+
+### `304` Not Modified status
+
+### Validity of date strings
+
+Talk about invalid date, last modified and expires, (0.11% of dates in desktop requests are invalid).
+
+Desktop - Valid date strings:
+99.18% using date, 72.57% using last-modified, 55.48% using expires
+
+Mobile - Valid date strings:
+99.07% using date, 70.55% using last-modified, 55.57% using expires
+
+Although 99% of both Desktop and Mobile pages use valid date strings, only 55% of Desktop and 56% of Mobile pages use 'expires'. Last-modified is used at a higher percentage, at 73% for Desktop and 71% for Mobile. While the amount of invalid 'date' and 'last-modified' directives is less than 1%, 2.75% of Mobile and 3.20% of Desktop 'expires' are being used incorrectly.
+
+What is or would be the impact from using expires more effectively?
+
+Could we possibly see big wins from getting more consistent here?
+
+### Vary
+
+Talk about Vary adoption, 42% of requests on mobile set Vary: accept-encoding.
+
+In addition, 83% of mobile Vary: accept-encoding requests also use cache-control. This is another case of the more you know about caching, the more consistently you can make use of all the resources available to you.
+
+The most common Vary headers are: `accept-encoding`, `user-agent`, `origin`, `accept`.
+
+### TTL adoption
+
+Talk about 28% of mobile requests use neither Cache-Control nor Expires headers.
+
+The median HTML resource on mobile has a TTL of 14 days
+
+## Cookie Cache adoption
+
+35.63% of Cacheable Desktop pages with set-cookie
+Mobile pages with set-cookie: 35.99%
+
+Another area of opportunity we discovered was with cacheable pages that are not using `set-cookie`. About 35% of Desktop and Mobile pages use `set-cookie` while over 64% of them do not.
+
+Large area of opportunity for pages that are actually cacheable, but are just missing this directive?
+
+Fun fact `test_cookie` is the most popular desktop and mobile cookie name at 4% of all cookies.
+
+{{ figure_markup(
+  caption="TODO",
+  content="4%",
+  classes="big-number",
+  sheets_gid="455647224",
+  sql_file="todo.sql"
+) }}
+
+The top cookies attributes used on Desktop and Mobile are SameSite and Secure. SameSite is on 53% of cookies set in Desktop responses and a whopping 68% on Mobile responses.
+
+Why is it a big deal that SameSite is at 68% on mobile?
+
+## What type of content are we caching?
+
+Talk about what kind of resources are we caching, (resource age groups: party, type).
+
```

Comment:
- Cacheable vs. non-cacheable responses
- Distribution of TTL in cacheable responses
- Cache TTL...? percentiles by resource type
- Distribution of cacheability by content type